### ISSUE TITLE
Naming of interior layers of the pretrained models

### ISSUE THREAD

**Issue by TreeLLi** [Microsoft/CNTK#3101]
Hello,

I was experimenting the effects of different base models on my network. But my program involves the using of functionalities like `graph.find_by_name()` which requires that I know the **naming of interior layers of the model**. So I am wondering where I can find such descriptions about the pretrained models, e.g. for the **ResNet-152**?

**Comment by KeDengMS** [comment]
You may use graph.plot() and find the names for the layers.

**Comment by TreeLLi** [comment]
Thanks for your help and it works. But the new trouble is that internal naming of your pretrained ResNet models are mess. I saw the **duplicated names** for different layers and the layers are named, e.g. `z.x.x.x.x.p` for a particular convolutional layer, by something **not human-readable** and not as they described in the paper.

**Comment by KeDengMS** [comment]
Yes the names are not quite user-friendly, as the model was trained from BrainScript. You may use [clone](https://www.cntk.ai/pythondocs/cntk.ops.functions.html?highlight=clone#cntk.ops.functions.Function.clone) to replace the parts you are interested in, so it would be easier to manipulate.
